### PR TITLE
New blog: Why we are writing SDKs

### PR DESCRIPTION
**Pull request by ukmadlz**
## Why
Blog: Why we are writing SDKs

### CODE REVIEW

**Review by LouManglass — COMMENTED** [review]
I'll get to these (#859, #860) later this afternoon.

File: website/blog/why-we-are-writing-sdks.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+title: "Why We Are Writing SDKs"
+author: Mike Elsmore
+author_url: "https://github.com/ukmadlz"
+author_image_url: "/img/team/mike-elsmore.jpg"
+category: Community
+---
+
+I’m new around here, and my first post is about something I really love… code that makes life easier!
+
+To begin with Optics CLI is ridiculously easy to start using within your application, you simply head to the root of your directory and run `api init`, configure your `.optic.yml` to wrap your application start command, and tada you’ve started gathering data about your ÅPI endpoints by just using them. Super simple and great right? To look at the full process and implement it yourself, please check out [our documentation](https://www.useoptic.com/docs/).
```
Comment (devdoshi) [inline website/blog/why-we-are-writing-sdks.md:12]: ÅPI -> API 😂

File: website/blog/why-we-are-writing-sdks.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+<!-- truncate -->
+
+## Introducing Optic SDKs
+Firstly, the idea behind these isn’t to be able to interact with Optics Cloud, or with the learning process of the CLI tool, it’s to simplify capturing the API traffic and sending the captures to the CLI. So when looking at it you won’t see handy API classes etc to interact with.
+
+With that out of the way, let’s take a look at what we have so far, it’s all very work in progress so please pass along your feedback so we can make it just what you want/need sooner rather than later.
+
+Firstly we put together a specification of what should be contained in the SDK, we’ve published this on notion for now to make commenting easier https://www.notion.so/useoptic/Optic-SDK-Specification-ff4d7ba6f0444c9eb0862a6d5748d707.
+What this specification tries to achieve is a language agnostic flow of what the SDK could do, from giivng a point of configuration, through to what it captures, the format is sends in, and where it sends the captured data.
```
Comment (devdoshi) [inline website/blog/why-we-are-writing-sdks.md:24]: giivng -> giving

File: website/blog/why-we-are-writing-sdks.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+But what happens when you can’t do that? Plenty of instances exist where it’s very complex to work in a team and wrap the command that starts your application. What happens when you use feature flags on that command, or you have to make operating system specific changes?
+
+<!-- truncate -->
+
+## Introducing Optic SDKs
+Firstly, the idea behind these isn’t to be able to interact with Optics Cloud, or with the learning process of the CLI tool, it’s to simplify capturing the API traffic and sending the captures to the CLI. So when looking at it you won’t see handy API classes etc to interact with.
+
+With that out of the way, let’s take a look at what we have so far, it’s all very work in progress so please pass along your feedback so we can make it just what you want/need sooner rather than later.
+
+Firstly we put together a specification of what should be contained in the SDK, we’ve published this on notion for now to make commenting easier https://www.notion.so/useoptic/Optic-SDK-Specification-ff4d7ba6f0444c9eb0862a6d5748d707.
```
Comment (devdoshi) [inline website/blog/why-we-are-writing-sdks.md:23]: - [ ] turn this into a markdown link

File: website/blog/why-we-are-writing-sdks.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+I’m new around here, and my first post is about something I really love… code that makes life easier!
+
+To begin with Optics CLI is ridiculously easy to start using within your application, you simply head to the root of your directory and run `api init`, configure your `.optic.yml` to wrap your application start command, and tada you’ve started gathering data about your ÅPI endpoints by just using them. Super simple and great right? To look at the full process and implement it yourself, please check out [our documentation](https://www.useoptic.com/docs/).
+
+But what happens when you can’t do that? Plenty of instances exist where it’s very complex to work in a team and wrap the command that starts your application. What happens when you use feature flags on that command, or you have to make operating system specific changes?
+
+<!-- truncate -->
+
+## Introducing Optic SDKs
+Firstly, the idea behind these isn’t to be able to interact with Optics Cloud, or with the learning process of the CLI tool, it’s to simplify capturing the API traffic and sending the captures to the CLI. So when looking at it you won’t see handy API classes etc to interact with.
```
Comment (devdoshi) [inline website/blog/why-we-are-writing-sdks.md:19]: so essentially we're calling them SDKs for Optic's developer tooling, with an implementation for each API framework

File: website/blog/why-we-are-writing-sdks.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+Firstly we put together a specification of what should be contained in the SDK, we’ve published this on notion for now to make commenting easier https://www.notion.so/useoptic/Optic-SDK-Specification-ff4d7ba6f0444c9eb0862a6d5748d707.
+What this specification tries to achieve is a language agnostic flow of what the SDK could do, from giivng a point of configuration, through to what it captures, the format is sends in, and where it sends the captured data. 
+
+Currently we have a work in progress Node SDK on GitHub (but not published to NPM) at https://github.com/opticdev/optic-node/tree/main/sdk, and this example just shows a working implementation of our SDK specification.
+## But this isn’t simpler than before?
+Yes reader, you’d be correct about that. It does allow for the more in code configuration of your captures and it does allow for a means to capture your API traffic within your application but it’s not plug and play. But this is step one in our master plan here at Optic to take over the world by being super simple, but that means I little more upfront work.
```
Comment (devdoshi) [inline website/blog/why-we-are-writing-sdks.md:28]: I little -> a little

File: website/blog/why-we-are-writing-sdks.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+Once we have these SDKs it allows for middleware to be produced simply and quickly, as all the hard work has been done up front in the SDK! If you are unaware of what middleware is, here is a one liner:
+
+> Middleware is a type of computer software that provides services to software applications beyond those available from the operating system. It can be described as "software glue".
+> https://en.wikipedia.org/wiki/Middleware
+
+What this means is we want to give you, our amazing users, a simple plug and play module in your desired framework be in Express in Node or Laravel in PHP, we want something you can quickly add to your existing tool chain and everyone working on the project have all the advantages of using Optic. We do have a couple of examples of what the middleware could be like for Express and Hapi in the Node world https://github.com/opticdev/optic-node/tree/main/frameworks, please feel free to comment or log issues that we can take and make it so much better for you.
```
Comment (devdoshi) [inline website/blog/why-we-are-writing-sdks.md:34]: - [ ] markdown links

File: website/blog/why-we-are-writing-sdks.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+## Feel like get involved?
```
Comment (devdoshi) [inline website/blog/why-we-are-writing-sdks.md:35]: get -> getting

File: website/blog/why-we-are-writing-sdks.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+Optic is only a small organization, we can’t possibly work in every language with every framework so we welcome the communities assistance in doing so. The amazing [Adrien Brault](https://github.com/adrienbrault) has already produced a proof of concept for PHP that you can see at https://github.com/adrienbrault/optic-php.
```
Comment (devdoshi) [inline website/blog/why-we-are-writing-sdks.md:36]: worth mentioning bounties?

File: website/blog/why-we-are-writing-sdks.md (hunk 4 above)
Comment (ukmadlz) [reply website/blog/why-we-are-writing-sdks.md:19]: And SDK per language, which ends up as a middleware for the frameworks yes.

File: website/blog/why-we-are-writing-sdks.md (hunk 8 above)
Comment (ukmadlz) [reply website/blog/why-we-are-writing-sdks.md:36]: I'd avoid that till we've worked out the kinks of how we want to do it

**Review by LouManglass — APPROVED** [review]
@ukmadlz  I made a few tweaks - changed the link text in a few links, added a few apostrophes, and swapped out what I call Manglass Link Format for Markdown where applicable. Feel free to revert anything (though the MLF tweaks should stay)

File: website/blog/why-we-are-writing-sdks.md [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+title: "Why We Are Writing SDKs"
+author: Mike Elsmore
+author_url: "https://github.com/ukmadlz"
+author_image_url: "/img/team/mike-elsmore.png"
+category: Community
+---
+
+I’m new around here, and my first post is about something I really love… code that makes life easier!
+
+To begin with Optic's CLI is ridiculously easy to start using within your application, you simply head to the root of your directory and run `api init`, configure your `.optic.yml` to wrap your application start command, and tada you’ve started gathering data about your API endpoints by just using them. Super simple and great right? To look at the full process and implement it yourself, please check out [our documentation](https://www.useoptic.com/docs/).
```
Comment (acunniffe) [inline website/blog/why-we-are-writing-sdks.md:12]: it's `optic.yaml`

no hidden and yeah we use the full spelling

File: website/blog/why-we-are-writing-sdks.md [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+But what happens when you can’t do that? Plenty of instances exist where it’s very complex to work in a team and wrap the command that starts your application. What happens when you use feature flags on that command, or you have to make operating system specific changes?
```
Comment (acunniffe) [inline website/blog/why-we-are-writing-sdks.md:14]: /....or when your API tests happen internally, not over the network

File: website/blog/why-we-are-writing-sdks.md [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+
+<!-- truncate -->
+
+## Introducing Optic SDKs
+Firstly, the idea behind these isn’t to be able to interact with Optic's Cloud, or with the learning process of the CLI tool, it’s to simplify capturing the API traffic and sending the captures to the CLI. So when looking at it you won’t see handy API classes etc to interact with.
+
+With that out of the way, let’s take a look at what we have so far, it’s all very work in progress so please pass along your feedback so we can make it just what you want/need sooner rather than later.
+
+Firstly we put together a specification of what should be contained in the SDK, we’ve published this on [Notion](https://www.notion.so/useoptic/Optic-SDK-Specification-ff4d7ba6f0444c9eb0862a6d5748d707) for now, please give it a look and add your comments.
```
Comment (acunniffe) [inline website/blog/why-we-are-writing-sdks.md:23]: this should move to useoptic.com/docs/....advanced -- let's discuss.

**Pull request merged**